src/jit/lowerarm64.cpp

```diff
@@ -1018,11 +786,352 @@ void Lowering::TreeNodeInfoInit(GenTree* stmt)
+
+    for (GenTreePtr list = call->gtCallLateArgs; list; list = list->MoveNext())
+    {
+        assert(list->IsList());
+
+        GenTreePtr argNode = list->Current();
+
+        fgArgTabEntryPtr curArgTabEntry = compiler->gtArgEntryByNode(call, argNode);
+        assert(curArgTabEntry);
+
+        if (curArgTabEntry->regNum == REG_STK)
+        {
+            // late arg that is not passed in a register
+            assert(argNode->gtOper == GT_PUTARG_STK);
+
+            TreeNodeInfoInitPutArgStk(argNode, curArgTabEntry);
+            continue;
+        }
+
+        var_types argType = argNode->TypeGet();
+        bool      argIsFloat = varTypeIsFloating(argType);
+        callHasFloatRegArgs |= argIsFloat;
+
+        regNumber argReg = curArgTabEntry->regNum;
+        // We will setup argMask to the set of all registers that compose this argument
+        regMaskTP argMask = 0;
+
+        argNode = argNode->gtEffectiveVal();
+
+        // A GT_LIST has a TYP_VOID, but is used to represent a multireg struct
+        if (varTypeIsStruct(argNode) || (argNode->gtOper == GT_LIST))
+        {
+            GenTreePtr actualArgNode = argNode;
+            unsigned originalSize = 0;
+
+            if (argNode->gtOper == GT_LIST)
+            {
+                // There could be up to 2-4 PUTARG_REGs in the list (3 or 4 can only occur for HFAs)
+                GenTreeArgList* argListPtr = argNode->AsArgList();
+
+                // Initailize the first register and the first regmask in our list
+                regNumber targetReg = argReg;
+                regMaskTP targetMask = genRegMask(targetReg);
+                unsigned iterationNum = 0;
+                originalSize = 0;
+
+                for (; argListPtr; argListPtr = argListPtr->Rest())
+                {
+                    GenTreePtr putArgRegNode = argListPtr->gtOp.gtOp1;
+                    assert(putArgRegNode->gtOper == GT_PUTARG_REG);
+                    GenTreePtr putArgChild = putArgRegNode->gtOp.gtOp1;
+
+                    originalSize += REGSIZE_BYTES;  // 8 bytes
+
+                                                    // Record the register requirements for the GT_PUTARG_REG node
+                    putArgRegNode->gtLsraInfo.setDstCandidates(l, targetMask);
+                    putArgRegNode->gtLsraInfo.setSrcCandidates(l, targetMask);
+
+                    // To avoid redundant moves, request that the argument child tree be 
+                    // computed in the register in which the argument is passed to the call.
+                    putArgChild->gtLsraInfo.setSrcCandidates(l, targetMask);
+
+                    // We consume one source for each item in this list
+                    info->srcCount++;
+                    iterationNum++;
+
+                    // Update targetReg and targetMask for the next putarg_reg (if any)
+                    targetReg = REG_NEXT(targetReg);
+                    targetMask = genRegMask(targetReg);
```

Comment:
This is ARM64 specific code and we are baking in the assumption that we have sequential argument registers.

I probably could use the more confusing genMapRegArgNumToRegNum() call here, but we don't have to deal with non-sequential argument registers on ARM32/ARM64.